consensus/src/lib.rs

```diff
@@ -384,13 +333,67 @@ impl<'a> Consensus<'a> {
         Ok(())
     }
 
-    fn is_block_in_main_chain(&self, block_index: &BlockIndex, storage_tx: &StoreTxRw) -> bool {
+    fn is_block_in_main_chain(&self, block_index: &BlockIndex) -> bool {
```

Comment:
Wouldn't a more appropriate name here be `is_tip_of_main_chain`?
The name as is suggest that the function checks whether the block with the given index is _anywhere_ on the main chain.
I may be missing something entirely, please correct me if so.

Reply:
I'd like to stay with `is_block_in_main_chain` because I check blocks, not only at the end of a chain. Have a look into `fn get_new_chain`. I collect block indexes there for a new chain until meet a common ancestor for the main chain and a new chain.

Reply:
I misunderstood on my first reading, now I understand, ignore my comment :)

Reply:
I'm resolving this because it seems done.